overlord/state/taskrunner.go

```diff
@@ -302,6 +326,22 @@ func (r *TaskRunner) Ensure() {
 	}
 }
 
+func (r *TaskRunner) cleanTask(t *Task) {
+	select {
+	case <-t.Change().Ready():
+	default:
+		return
+	}
+	cleanup, ok := r.cleanups[t.Kind()]
+	if ok {
+		if err := cleanup(t); err != nil {
```

Comment:
Changed cleanup handlers to be typical, working just like any other handler. This is the most flexible as it won't force timely execution, and the least error prone as we're used to other handlers already.